crates/rpc/src/jsonrpc/router/subscription.rs

```diff
@@ -197,8 +198,15 @@ where
             _phantom: Default::default(),
         };
 
-        let starting_block = T::starting_block(&params);
+        let max_history = router
+            .context
+            .websocket
+            .as_ref()
+            .map(|ws_ctx| ws_ctx.max_history)
+            .expect("Subscription methods should not be called when websockets are disabled");
```

Comment:
I tried that (by accident), and it turns out this code can be called even with false `rpc.websocket.enabled`; probably that could be fixed, but in the meantime (and for the next time), IMHO this should be a recoverable failure...